tests/ports.py

```diff
@@ -0,0 +1,119 @@
+"""
+Test Port mappings work on running non-jupyter workflows
+"""
+import subprocess
+import urllib3
+import time
+import os
+import tempfile
+import signal
+import random
+
+
+def read_port_mapping_response(host, port, protocol = None):
+    """
+    Deploy container and test if port mappings work as expected
+
+    Args:
+        host: the host interface to bind to.
+        port: the random host port to bind to
+        protocol: the protocol to use valid values /tcp or /udp
+    """
+    builddir = os.path.dirname(__file__)
+    port_protocol = '8000'
+    if protocol:
+        port_protocol += protocol
+    host_port = port
+    if host:
+        host_port = host + ':' + port
+    else:
+        host = 'localhost'
+    with tempfile.TemporaryDirectory() as tmpdir:
+        username = os.getlogin()
+
+        proc = subprocess.Popen(['repo2docker',
+                                 '-p',
```

Comment:
The other option here would be to figure out a way to specify the container name in repo2docker, but I think for now this is sufficient. Perhaps add a comment explaining why this is here?

Reply:
Could you explain a bit more on what I need to explain? Is it the part about the volumes?

Reply:
Just a general comment explaining what is going on here :) We're doing a bunch of things - using a volume to signal when something is done building, keeping a child process running (and killing it after, etc).

Reply:
Ah ok ...